### PR TITLE
Feat: gateway trait support https protocol (#3832)

### LINKED ISSUE #3832: [Feature] gateway trait should support https protocol
**Is your feature request related to a problem? Please describe.**

The ingress behind of the gateway trait support  tls configuration but we didn't expose the arguments.

https://kubernetes.io/docs/concepts/services-networking/ingress/#tls

We should support it to support https route.

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by oiar**
### Description of your changes
The gateway trait support https protocol (#3832)

Fixes #3832

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [x] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

**Comment by fourierr** [comment]
please run make renewable locally and add DCO before push

File: charts/vela-core/templates/defwithtemplate/gateway.yaml [hunk 1]
```diff
@@ -44,6 +44,16 @@ spec:
         		if parameter.classInSpec {
         			ingressClassName: parameter.class
         		}
+        		if parameter.tlsInSpec {
```
Comment (wonderflow) [inline charts/vela-core/templates/defwithtemplate/gateway.yaml:47]: what about use `secretName` to check if we use tls or not, then we don't need a redundant `tlsInSpec` field.

```suggestion
        		if parameter.secretName != _|_ {
```

**Comment by fourierr** [comment]
ltgm, 
- if we extend tls to slice, the parameter.secretName also need to extend to slice, 
- the users who use the former version will be affected. 
- To keep compatibility,we can extend it after cue definition revisions
@wonderflow @oiar

**Review by fourierr — APPROVED** [review]

**Comment by oiar** [comment]
> generally LGTM. @oiar you can refer to @fourierr suggustion, but we should keep the user experience clear

The rules only support one, now need to support multiple tls? The hosts in the tls section need to explicitly match the host in the rules section. Or do we need to modify the rules to be multiple?

File: pkg/controller/core.oam.dev/v1alpha2/application/application_controller_test.go [hunk 2]
```diff
@@ -2664,6 +2689,59 @@ var _ = Describe("Test Application Controller", func() {
...
+			return k8sClient.Get(ctx, client.ObjectKey{Name: app.Name + "-v1", Namespace: app.GetNamespace()}, appRev)
+		}, 10*time.Second, 500*time.Millisecond).Should(Succeed())
+
+		By("Check ingress Created with the expected trait-gateway spec")
+		ingress := &v1beta12.Ingress{}
+		Expect(k8sClient.Get(ctx, client.ObjectKey{
+			Namespace: ns.Name,
+			Name:      app.Spec.Components[0].Name + "-ingress",
+		}, ingress)).Should(BeNil())
+
```
Comment (wonderflow) [inline pkg/controller/core.oam.dev/v1alpha2/application/application_controller_test.go:2740]: why don't check the ingress has the secret and tls fields here?  It's the main goal for the long test, but you ignored.

**Pull request merged**